src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -475,6 +496,51 @@ public bool TryGetJsonObjectPropertyValue(string propertyName, out JsonObject js
             return false;
         }
 
+        /// <summary>
+        ///   Returns the JSON array value of a property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <returns>JSON objectvalue of a property with the specified name.</returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON array.
+        /// </exception>
+        /// <exception cref="InvalidCastException">
+        ///   Property with specified name is not a JSON array.
+        /// </exception>
+        public JsonArray GetJsonArrayPropertyValue(string propertyName)
+        {
+            if (GetPropertyValue(propertyName) is JsonArray jsonArray)
+            {
+                return jsonArray;
+            }
+
+            throw new InvalidCastException(SR.Format(SR.PropertyTypeMismatch, propertyName));
```

Comment:
Don't throw InvalidCastException manually.  This should presumably be an ArgumentException.